idea-plugin/src/main/java/com/palantir/gradle/jdks/PluginUpdateCheckerService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.intellij.externalDependencies.DependencyOnPlugin;
+import com.intellij.externalDependencies.ExternalDependenciesManager;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationGroupManager;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.PluginsAdvertiser;
+import com.intellij.util.text.VersionComparatorUtil;
+import java.util.Optional;
+import java.util.Set;
+
+@Service(Service.Level.PROJECT)
+public final class PluginUpdateCheckerService {
+
+    private static final String PLUGIN_ID = "palantir-gradle-jdks";
+    private final Logger logger = Logger.getInstance(PluginUpdateCheckerService.class);
+    private final Project project;
+
+    public PluginUpdateCheckerService(Project project) {
+        this.project = project;
+    }
+
+    public void checkPluginVersion() {
+        PluginId pluginId = PluginId.getId(PLUGIN_ID);
+        IdeaPluginDescriptor pluginDescriptor = PluginManagerCore.getPlugin(pluginId);
+        if (pluginDescriptor == null) {
+            logger.info("Plugin " + pluginId + " not found");
+            return;
+        }
+        Optional<String> maybeMinVersion =
+                ExternalDependenciesManager.getInstance(project).getDependencies(DependencyOnPlugin.class).stream()
+                        .filter(dependencyOnPlugin ->
+                                dependencyOnPlugin.getPluginId().equals(pluginId.getIdString()))
+                        .map(DependencyOnPlugin::getMinVersion)
```

Comment:
Just thought of a possible complication:

1. We release a new version of a gradle + intellij plugin, set the min-version of intellij plugin to "force" people to upgrade.
2. The Gradle plugin update immediately gets picked up by excavator, applied to people's repos
3. Jetbrains takes two business days to approve our intellij plugin.
4. Every single time Gradle is refreshed, people get the popup to update to the new IntelliJ version that _is not available yet_.

Reply:
that's a good point, I'll see if I can check that the version exists before opening the popup.
But that means we cannot really rely on the minVersion. Probably the best would be if the plugin will always maintain backwards compatibility with the intellij plugin and I can add a test for the logic [here](https://github.com/palantir/gradle-jdks/blob/develop/gradle-jdks/src/main/java/com/palantir/gradle/jdks/PalantirGradleJdksIdeaPlugin.java#L28) that we won't update the plugin until it actually exists.

Reply:
added a test that checks that the min-version intellij plugin: https://github.com/palantir/gradle-jdks/pull/410/files#diff-1d01bc54a43b33238be402e57110068b45d10d5b2c7f88104b1c1e67080c1a53R35. This should stop us from bumping the required min-version in externalDependencies.xml file.

I could add the check (probably cached + expiration) [here](https://github.com/palantir/gradle-jdks/pull/410/files#diff-c40f3e7fde707d4f846f1cb40c95c063ac736bb45bc484fa9c9b712dfa537c5aR67) in case we are going to "recall"(delete) a version, but I am not sure if we need it atm.